src/rred_reports/masterfile.py

```diff
@@ -142,5 +153,13 @@ def join_masterfile_dfs(masterfile_dfs: dict[str, pd.DataFrame]) -> pd.DataFrame
     Returns:
         pd.DataFrame: joined masterfile
     """
-    pupil_teachers = pd.merge(masterfile_dfs["pupils"], masterfile_dfs["teachers"], on="rred_user_id")
-    return pd.merge(pupil_teachers, masterfile_dfs["schools"], on="school_id")
+    teacher_schools = pd.merge(masterfile_dfs["teachers"], masterfile_dfs["schools"], on="school_id")
+    return pd.merge(teacher_schools, masterfile_dfs["pupils"], on="rred_user_id")
+
+
+def masterfile_columns():
+    pupil_no, user_id, *other_pupil_fields = Pupil.fields()
+    school_id, *other_school_fields = School.fields()
+    user_id, *other_teacher_fields, school_id = Teacher.fields()
```

Comment:
do we want to return `school_id` too?

Reply:
yeah TBF one of the `school_id`s are just ignored, but it was to be specific about what fields they were. could always just use `_` here instead to make it explicit that we don't care about it

Reply:
oh now I see what you meant, yeah definitely did want to return that 🤦